CMakeLists.txt

```
@@ -31,8 +31,11 @@ if(NOT PYTHON_EXECUTABLE)
   message(FATAL_ERROR "Missing PYTHON_EXECUTABLE")
 endif()
 
+set(GEN_SRC "${CMAKE_CURRENT_BINARY_DIR}/generated-sources")
+
 set(XPP_INCLUDE_DIRS
   ${PROJECT_SOURCE_DIR}/include
+  ${GEN_SRC}/include
```

```suggestion
  ${GEN_SRC}/include/xpp
```
If you modify this variable, it should work without modifying the includes.

---

But I want clients of this library to still do includes via `xpp/...` because I don't want to pollute the include file namespace.

---

Hum indeed

---

There probably isn't a way to do this anyway

---

There is a way to do that but by looking at the directory structure of this repo, I think this is better this way.